monai/transforms/compose.py

```diff
@@ -73,16 +74,19 @@ def evaluate_with_overrides(
         override: keyword arguments to apply transforms.
         override_keys: to which the override arguments are used when apply transforms.
         verbose: whether to print debugging info when evaluate MetaTensor with pending operations.
+        mode: "default" or "strict".
 
     """
     if not lazy_evaluation:
         return data  # eager evaluation
     overrides = (overrides or {}).copy()
     if isinstance(data, monai.data.MetaTensor):
+        if isinstance(upcoming, mt.MapTransform) and override_keys not in upcoming.keys:
+            return data
         if data.has_pending_operations and (
             (upcoming is None)
-            or (isinstance(upcoming, mt.Identity))
-            or (isinstance(upcoming, mt.Identityd) and override_keys in upcoming.keys)
+            or (isinstance(upcoming, (mt.Identity, mt.Identityd)))
+            or (mode == "strict" and not isinstance(upcoming, mt.LazyTransform))
```

Comment:
This change fixes precisely one aspect of the issues raised. `mode` has no consideration of extensibility. "strict" should be the default mode; the other mode should be called "benchmark" or "relaxed". There is no discussion in any documentation about the modes and their execution behaviour. This function is still far too complex and only maintainable by the person who wrote it. The Compose API still has all of the issues raised in #6257.